core/src/main/scala/memeid/UUID.scala

```diff
@@ -138,128 +138,170 @@ sealed trait UUID extends Comparable[UUID] {
+    * @param lsb Least significant bit in [[scala.Long Long]] representation
+    * @return a new [[UUID]] constructed from msb and lsb
+    */
   def from(msb: Long, lsb: Long): UUID = new JUUID(msb, lsb).asScala
 
   /**
-   * Creates a [[UUID UUID]] from the [[java.util.UUID#toString string standard representation]]
-   * wrapped in a [[scala.util.Right Right]].
-   *
-   * Returns [[scala.util.Left Left]] with the error in case the string doesn't follow the
-   * string standard representation.
-   *
-   * @param s String for the [[java.util.UUID UUID]] to be generated as an [[UUID]]
-   * @return [[scala.util.Either Either]] with [[scala.util.Left Left]] with the error in case the string doesn't follow the
-   *        string standard representation or [[scala.util.Right Right]] with the [[UUID UUID]] representation.
-   */
-  def from(s: String): Either[Throwable, UUID] = Try(JUUID.fromString(s).asScala).toEither
+    * Creates a [[UUID UUID]] from the [[java.util.UUID#toString string standard representation]]
+    * wrapped in a [[scala.util.Right Right]].
+    *
+    * Returns [[scala.util.Left Left]] with the error in case the string doesn't follow the
+    * string standard representation.
+    *
+    * @param s String for the [[java.util.UUID UUID]] to be generated as an [[UUID]]
+    * @return [[scala.util.Either Either]] with [[scala.util.Left Left]] with the error in case the string doesn't follow the
+    *        string standard representation or [[scala.util.Right Right]] with the [[UUID UUID]] representation.
+    */
+  def from(s: String): Either[Throwable, UUID] =
+    Try(JUUID.fromString(s).asScala).toEither
 
   /**
-   * The nil UUID is special form of UUID that is specified to have all 128 bits set to zero.
-   *
-   * @see [[https://tools.ietf.org/html/rfc4122#section-4.1.7]]
-   */
+    * The nil UUID is special form of UUID that is specified to have all 128 bits set to zero.
+    *
+    * @see [[https://tools.ietf.org/html/rfc4122#section-4.1.7]]
+    */
   case object Nil extends UUID { override val juuid: JUUID = new JUUID(0, 0) }
 
   /**
-   * Version 1 UUIDs are those generated using a timestamp and the MAC address of the
-   * computer on which it was generated.
-   *
-   * @see [[https://tools.ietf.org/html/rfc4122#section-4.1.3]]
-   */
-  final class V1 private[memeid] (override private[memeid] val juuid: JUUID) extends UUID
+    * Version 1 UUIDs are those generated using a timestamp and the MAC address of the
+    * computer on which it was generated.
+    *
+    * @see [[https://tools.ietf.org/html/rfc4122#section-4.1.3]]
+    */
+  final class V1 private[memeid] (override private[memeid] val juuid: JUUID)
+      extends UUID {
+
+    /**
+      * Get the time_low component of the timestamp field
+      * @see [[https://tools.ietf.org/html/rfc4122#section-4.1.2]]
+      * @return time_low component of the timestamp field
+      */
+    def timeLow: Long = readByte(mask(32, 0), juuid.timestamp())
```

Comment:
Now that we use these masks when setting and retrieving time components we may want to store them in a val statically like the version mask.